base/util_testing.go

```diff
@@ -222,8 +238,13 @@ func (tbm *TestBucketManager) OpenTestBucket() (bucketExists bool, err error) {
 	username, password, _ := tbm.BucketSpec.Auth.GetCredentials()
 	bucket, err := tbm.Cluster.OpenBucket(tbm.BucketSpec.BucketName, password)
 	if err != nil {
-		// Let's assume that if there is an error opening the bucket, it's just because the
-		// bucket does not exist
+		// Authentication failure should return an explicit error as we can't continue from here.
+		if pkgerrors.Cause(err) == gocb.ErrAuthError {
+			log.Printf("Unable to authenticate as %s: %v", username, err)
+			return false, err
```

Comment:
Should this be returning 'true, err'?

Reply:
I don't think so. If gocb can't authenticate on connect, it will immediately return, so we don't know if a bucket exists or not.

Consumers of this method check the error first, before attempting to use the bucketExists value. It's probably safer to return false.

Reply:
Makes sense - thanks.